pandas/core/indexing.py

```diff
@@ -162,7 +163,8 @@ def iloc(self) -> _iLocIndexer:
         - A ``callable`` function with one argument (the calling Series or
           DataFrame) and that returns valid output for indexing (one of the above).
           This is useful in method chains, when you don't have a reference to the
-          calling object, but would like to base your selection on some value.
+          calling object, but would like to base your selection on
+          some value. Note that returning a tuple from a callable is deprecated.
```

Comment:
The deprecation message should be in a separate section starting with `.. deprecated:: 2.1.0`. Look through the code base for examples.